testApi/src/main/scala/africa/absa/testing/scapi/rest/response/LogResponseAction.scala

```diff
@@ -54,10 +57,13 @@ object LogResponseAction extends ResponsePerformer {
    * @throws IllegalArgumentException if the response action's name is not recognized.
    */
   def performResponseAction(response: Response, responseAction: ResponseAction): Boolean = {
+    val message = responseAction.params.getOrElse("message", throw new IllegalArgumentException("Missing 'message' parameter"))
+
     responseAction.name match {
-      case INFO =>
-        val message = responseAction.params("message")
-        logInfo(message)
+      case ERROR => logError(message)
+      case WARN => logWarn(message)
+      case INFO => logInfo(message)
+      case DEBUG => logDebug(message)
```

Comment:
Again, these should be probably enum/sealed case objects. You would get match exhaustion checking from the compiler with the sealed case object, e.g.

Reply:
Fixed in commit 2e251aeeb8d90762628f8d6456e2a9c08e9d882d.